File: SYP.Web/Modules/Order/Order/OrderForm.cs
using Serenity.ComponentModel;
using System;

namespace SYP.Order.Forms;

[FormScript("Order.Order")]
[BasedOnRow(typeof(OrderRow), CheckNames = true)]
public class OrderForm
{
    [Tab("Genel Bilgiler")]
    [Category("Sipariş Bilgileri")]
    [HalfWidth, ReadOnly(true), Placeholder("Otomatik oluşturulacak")]
    public string OrderNumber { get; set; }

    [HalfWidth]
    public int CustomerId { get; set; }

    [HalfWidth, ReadOnly(true)]
    public int ManagerUserId { get; set; }

    [HalfWidth]
    public DateTime OrderDate { get; set; }

    [HalfWidth]
    public OrderStatus Status { get; set; }

    [HalfWidth]
    public int CurrencyId { get; set; }

    [Category("Tutar Bilgileri")]
    [HalfWidth, ReadOnly(true), DisplayFormat("#,##0.00")]
    public decimal TotalAmount { get; set; }

    [HalfWidth, ReadOnly(true), DisplayFormat("#,##0.00")]
    public decimal DiscountPercentage { get; set; }

    [HalfWidth, ReadOnly(true), DisplayFormat("#,##0.00")]
    public decimal DiscountAmount { get; set; }

    [HalfWidth, ReadOnly(true), DisplayFormat("#,##0.00")]
    public decimal NetAmount { get; set; }

    [Tab("Sipariş Kalemleri")]
    [MasterDetailRelation(foreignKey: "OrderId"), DisplayName("Sipariş Kalemleri")]
    public List<OrderDetailRow> DetailList { get; set; }

    [Tab("Notlar ve Açıklamalar")]
    [Category("Notlar")]
    [FullWidth, TextAreaEditor(Rows = 5)]
    public string Notes { get; set; }

    [Category("Red/İptal Nedeni")]
    [FullWidth, TextAreaEditor(Rows = 3)]
    [Hint("Sipariş reddedildiğinde veya iptal edildiğinde zorunludur")]
    public string RejectReason { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Order/OrderForm.ts
import { DateTimeEditor, DecimalEditor, EnumEditor, initFormType, LookupEditor, PrefixedContext, ServiceLookupEditor, StringEditor, TextAreaEditor } from "@serenity-is/corelib";
import { OrderDetailGridEditor } from "../../Order/OrderDetail/Editor/OrderDetailGridEditor";
import { OrderStatus } from "./OrderStatus";

export interface OrderForm {
    OrderNumber: StringEditor;
    CustomerId: ServiceLookupEditor;
    ManagerUserId: LookupEditor;
    OrderDate: DateTimeEditor;
    Status: EnumEditor;
    CurrencyId: LookupEditor;
    TotalAmount: DecimalEditor;
    DiscountPercentage: DecimalEditor;
    DiscountAmount: DecimalEditor;
    NetAmount: DecimalEditor;
    DetailList: OrderDetailGridEditor;
    Notes: TextAreaEditor;
    RejectReason: TextAreaEditor;
}

export class OrderForm extends PrefixedContext {
    static readonly formKey = 'Order.Order';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!OrderForm.init) {
            OrderForm.init = true;

            var w0 = StringEditor;
            var w1 = ServiceLookupEditor;
            var w2 = LookupEditor;
            var w3 = DateTimeEditor;
            var w4 = EnumEditor;
            var w5 = DecimalEditor;
            var w6 = OrderDetailGridEditor;
            var w7 = TextAreaEditor;

            initFormType(OrderForm, [
                'OrderNumber', w0,
                'CustomerId', w1,
                'ManagerUserId', w2,
                'OrderDate', w3,
                'Status', w4,
                'CurrencyId', w2,
                'TotalAmount', w5,
                'DiscountPercentage', w5,
                'DiscountAmount', w5,
                'NetAmount', w5,
                'DetailList', w6,
                'Notes', w7,
                'RejectReason', w7
            ]);
        }
    }
}

[OrderStatus]; // referenced types